File: C#/Template/PdfCVReportGeneration.cs
using Template.Models;

namespace Template;

public sealed class PdfCVReportGeneration : CvReportGeneration
{
    /// <summary>
    /// PDF Scanning implementation
    /// </summary>
    /// <param name="cvPath"><inheritdoc/></param>
    /// <returns><inheritdoc/></returns>
    protected override Data ExtractData(string cvPath)
    {
        Console.WriteLine("Extracting data from pdf...");
        return new Data();
    }
}

File: Python/template/PdfCVReportGeneration.py
from CVReportGeneration import CVReportGeneration
from io import FileIO as File
from Data import Data

class PdfCVReportGeneration(CVReportGeneration):
    def __init__(self):
        super().__init__()

    def _extract_data(self, file: File) -> Data:
        print(f"Extracting data from a PDF file...")
        return None